listener.go

```diff
@@ -38,7 +38,7 @@ func Listen(network string, laddr *net.UDPAddr, config *Config) (net.Listener, e
 }
 
 // NewListener creates a DTLS listener which accepts connections from an inner Listener.
-func NewListener(inner net.Listener, config *Config) (net.Listener, error) {
+func NewListener(inner net.Listener, config *Config) (*listener, error) { //nolint:golint
```

Comment:
What was your thinking for the use case here? We don't acutally use `NewListener` in mp-relay right now, instead we call the `Listen` function at the top of this file.

Are we going to pass in our own parent listener with a custom `AcceptFilter` set?

Reply:
Yes I think that was the intent.

Reply:
After more investigation, I don't think we can use the AcceptFilter for this.

For now, the filtering we want to do should be done on our side, after Accept has given us a Conn. Earlier filtering will require forking the UDP package so that the remote address is given to the accept filter in (*udp.listener).getConn